### PR TITLE
Failing tests for the identifier parser

### PR DESCRIPTION
**Pull request by sjakobi**
While extending the ~~identifier parser~~ identifier parser tests I found the following test cases that I thought should pass.

If someone can confirm that they should pass, I'll try to fix the parser.

### CODE REVIEW

File: haddock-library/test/Documentation/Haddock/ParserSpec.hs [hunk 1]
```diff
@@ -86,6 +86,9 @@ spec = do
       it "parses identifiers enclosed within backticks" $ do
         "`foo`" `shouldParseTo` DocIdentifier "foo"
 
+      it "doesn't parse a composed function as an identifier" $ do
+        "'foo.bar'" `shouldParseTo` "'foo.bar'"
```
Comment (harpocrates) [inline haddock-library/test/Documentation/Haddock/ParserSpec.hs]: This looks potentially problematic: qualified identifiers should still be identified as `DocIdentifier`, no?

Comment (alexbiehl) [reply haddock-library/test/Documentation/Haddock/ParserSpec.hs]: But this is not a valid and qualified  haskell identifier, right?

File: haddock-library/test/Documentation/Haddock/ParserSpec.hs [hunk 2]
```diff
@@ -99,9 +102,21 @@ spec = do
       it "doesn't parse empty identifiers" $ do
         "``" `shouldParseTo` "``"
 
-      it "can parse infix identifiers" $ do
+      it "can parse an identifier in infix notation enclosed within backticks" $ do
         "``infix``" `shouldParseTo` "`" <> DocIdentifier "infix" <> "`"
 
+      it "can parse identifiers in infix notation enclosed within single quotes" $ do
+        "'`infix`'" `shouldParseTo` "`" <> DocIdentifier "infix" <> "`"
```
Comment (sjakobi) [inline haddock-library/test/Documentation/Haddock/ParserSpec.hs]: I think this is the most problematic of all of these test failures, as exactly this syntax is described http://haskell-haddock.readthedocs.io/en/latest/markup.html#hyperlinked-identifiers.

Here's the test output:

       expected: DocAppend (DocString "`") (DocAppend (DocIdentifier "infix") (DocString "`"))
        but got: DocAppend (DocString "'") (DocAppend (DocIdentifier "infix") (DocString "'"))

Comment (sjakobi) [reply haddock-library/test/Documentation/Haddock/ParserSpec.hs]: I just saw that this has been reported in https://github.com/haskell/haddock/issues/780.

**Comment by harpocrates** [comment]
@sjakobi It would be nice to rebase this onto the 8.8 branch. I'm pretty sure almost all of these would now pass (except perhaps `'foo.bar'` - but even fixing that should be now be pretty easy).

If you want to wait until after 8.8 is released, that's OK too (building all the deps is a pain...).

**Comment by Kleidukos** [comment]
Hi, thank you for this PR, but Haddock now lives full-time in the GHC repository! 
Read more at https://discourse.haskell.org/t/haddock-now-lives-in-the-ghc-repository/9576.

Let me know if you feel it is still needed, and I'll migrate it. :)

**Pull request closed without merging**